src/Symfony/Component/Form/CHANGELOG.md

```diff
@@ -9,6 +9,7 @@ CHANGELOG
  * deprecated the "cascade_validation" option in favor of setting "constraints"
    with the Valid constraint
  * moved data trimming logic of TrimListener into StringUtil
+ * [BC BREAK] When registering a type extension through the DI extension, the tag alias has to match the actual extended type.
```

Comment:
the service id makes no sense as there could only be one extension for a type (as the service ids need to be unique). So IMO this fallback really needs to be deprecated, making the "alias" required in the future.